pkg/plugin/plugin.go

```diff
@@ -86,9 +86,8 @@ func (r *RpcPlugin) SetWeight(
 	slog.Debug("the services name", slog.String("stable", stableSvcName), slog.String("canary", canarySvcName))
 
 	// TODO: filter by condition(s)
-	services := utils.Must1(getServiceList(httpProxy.Spec.Routes))
-	canarySvc := utils.Must1(getService(canarySvcName, services))
-	stableSvc := utils.Must1(getService(stableSvcName, services))
+	canarySvc := utils.Must1(getService(canarySvcName, &httpProxy))
```

Comment:
if you like remove the `getServiceList`, then search the `canaryService & stableService` at the same time , maybe better?

Reply:
Instead of removing the `getServiceList`, I modified it to return a map instead of a list to avoid iterating over the HTTPProxy's service list.